### ISSUE TITLE
Avocado server test FAILed

### ISSUE THREAD

**Issue by lmr** [avocado-framework/avocado-server#19]
python selftests/all/functional/avocadoserver/api.py
After following the instructions of the `avocado-run-testplan` tool, here's what I've found:

    Profilers declared: ['vmstat 1', 'journalctl -f']
    Profiler disabled
    START selftests/all/functional/avocadoserver/api.py

    Test instance parameters:
        id = selftests/all/functional/avocadoserver/api.py
        omit_non_tests = False

    Default parameters:
        base_url = http://127.0.0.1:9405
        password = 123
        username = admin

    Test instance params override defaults whenever available

    Testing that the server returns its version

    Traceback (most recent call last):
      File "selftests/all/functional/avocadoserver/api.py", line 246, in action
        self.test_version()
      File "selftests/all/functional/avocadoserver/api.py", line 49, in test_version
        self.get("/version/")
      File "selftests/all/functional/avocadoserver/api.py", line 30, in get
        self.assertEquals(response.status_code, status_code)
      File "/usr/lib64/python2.7/unittest/case.py", line 551, in assertEqual
        assertion_func(first, second, msg=msg)
      File "/usr/lib64/python2.7/unittest/case.py", line 544, in _baseAssertEqual
        raise self.failureException(msg)
    AssertionError: 403 != 200

    FAIL selftests/all/functional/avocadoserver/api.py -> AssertionError: 403 != 200

**Comment by lmr** [comment]
Test 3/4: Avocado Server Development Run
    Description: On a fresh machine checkout a copy of the avocado-server repository. Then run the quick and dirty script to get it running ./reset-and-run.sh. Expected result: "Starting development server at http://0.0.0.0:9405/"

    Result ([P]ass, [F]ail, [S]kip): P
    Additional Notes: It was hard to install all the dependencies

    Test 4/4: Avocado Server Functional Test
    Description: Run the selftests/all/functional/avocadoserver/api.py on the previously started Avocado Server

**Comment by clebergnu** [comment]
It looks like this is something related to the environment (we have to narrow it down). On my machine:

```
$ python selftests/all/functional/avocadoserver/api.py
Profilers declared: ['vmstat 1', 'journalctl -f']
Profiler disabled
START selftests/all/functional/avocadoserver/api.py

Test instance parameters:
    id = selftests/all/functional/avocadoserver/api.py
    omit_non_tests = False

Default parameters:
    base_url = http://127.0.0.1:9405
    password = 123
    username = admin

Test instance params override defaults whenever available

Testing that the server returns its version
Testing that the server has preloaded job statuses
Testing that the server does not allow adding a new job status
Testing that the server has preloaded test statuses
Testing that the server does not allow adding a new test status
Testing that the server has preloaded software component kinds
Testing that the server has preloaded software component arches
Testing that the server responds to software component listing
Testing that the server adds a software component
Testing that the server responds to linux distro listing
Testing that the server adds a linux distro
Testing that the server does not add a duplicated linux distro
Testing that the server has no test environments
Testing that the server adds a test environment
Testing that the server has no jobs
Testing that a new job can be added
Testing that a job can be deleted
Deleting job using path: /jobs/a0a272a09d2edda895bae4d75f5aebfad6562fb0/
Testing that the server has no jobs
Testing that a newly added job has no activities
Testing that a new job can be added
Testing that a job can be deleted
Deleting job using path: /jobs/a0a272a09d2edda895bae4d75f5aebfad6562fb0/
Testing that the server has no jobs
Not logging /var/log/messages (lack of permissions)
PASS selftests/all/functional/avocadoserver/api.py
```

**Comment by lmr** [comment]
I'm guessing the problem here might have to do with restframework versions.

**Comment by clebergnu** [comment]
Now that I've updated my development machine I also get this error:

```
$ avocado run selftests/all/functional/avocadoserver/api.py 
JOB ID     : 5cde3ad1630b907ed95862048e7dee12336b2cd0
JOB LOG    : /home/cleber/avocado/job-results/job-2015-06-07T23.25-5cde3ad/job.log
JOB HTML   : /home/cleber/avocado/job-results/job-2015-06-07T23.25-5cde3ad/html/results.html
TESTS      : 20
(1/20) selftests/all/functional/avocadoserver/api.py:api.test_jobpriority_list: PASS (0.13 s)
(2/20) selftests/all/functional/avocadoserver/api.py:api.test_jobpriority_noadd: FAIL (0.04 s)
(3/20) selftests/all/functional/avocadoserver/api.py:api.test_jobs_activities_empty: PASS (0.50 s)
(4/20) selftests/all/functional/avocadoserver/api.py:api.test_jobs_add: PASS (0.10 s)
(5/20) selftests/all/functional/avocadoserver/api.py:api.test_jobs_del: PASS (0.25 s)
(6/20) selftests/all/functional/avocadoserver/api.py:api.test_jobs_empty: PASS (0.13 s)
(7/20) selftests/all/functional/avocadoserver/api.py:api.test_jobstatus_list: PASS (0.14 s)
(8/20) selftests/all/functional/avocadoserver/api.py:api.test_jobstatus_noadd: PASS (0.14 s)
(9/20) selftests/all/functional/avocadoserver/api.py:api.test_linuxdistro_add: PASS (0.33 s)
(10/20) selftests/all/functional/avocadoserver/api.py:api.test_linuxdistro_list: PASS (0.13 s)
(11/20) selftests/all/functional/avocadoserver/api.py:api.test_linuxdistro_no_add_dup: PASS (0.19 s)
(12/20) selftests/all/functional/avocadoserver/api.py:api.test_softwarecomponent_add: PASS (0.23 s)
(13/20) selftests/all/functional/avocadoserver/api.py:api.test_softwarecomponent_list: PASS (0.11 s)
(14/20) selftests/all/functional/avocadoserver/api.py:api.test_softwarecomponentarch_list: PASS (0.14 s)
(15/20) selftests/all/functional/avocadoserver/api.py:api.test_softwarecomponentkind_list: PASS (0.17 s)
(16/20) selftests/all/functional/avocadoserver/api.py:api.test_testenvironment_add: PASS (0.32 s)
(17/20) selftests/all/functional/avocadoserver/api.py:api.test_testenvironment_empty: FAIL (0.14 s)
(18/20) selftests/all/functional/avocadoserver/api.py:api.test_teststatus_list: PASS (0.14 s)
(19/20) selftests/all/functional/avocadoserver/api.py:api.test_teststatus_noadd: PASS (0.13 s)
(20/20) selftests/all/functional/avocadoserver/api.py:api.test_version: PASS (0.13 s)
PASS       : 18
ERROR      : 0
FAIL       : 2
SKIP       : 0
WARN       : 0
INTERRUPT  : 0
TIME       : 3.57 s
```

I'm going to send a fix for it.

**Comment by clebergnu** [comment]
Fixed on #32
